### DISCUSSION TITLE
pgplot missing?

### DISCUSSION THREAD

**Discussion by walkloud** [Homebrew/discussions#911] [Everyday usage] [answered]
The internet believes `pgplot` is brew-installalbe, but when I type `brew info pgplot` or `brew install pgplot` I get 
```
Error: No available formula or cask with the name "pgplot".
```
Similarly, searching the gitrepo
```
https://github.com/Homebrew/homebrew-core/tree/master/Formula
```
I can not find pgplot.  The internet (via Google) has not provided any clues if it was dropped for some reason.  I'm trying to help a colleague get some gfortran code working that relies upon pgplot - he had to upgrade his OS to Big Sur, and so we are working on upgrading his homebrew.  I can see he has a remnant pgplot from an pre-updated homebrew in /usr/local, but we're stuck on upgrading given the above troubles.

Any help would be appreciated.

**Comment by jonchang** [comment, marked as answer]
`pgplot` was deleted due to having licensing issues. You may be interested in giza which aims to be a free replacement for pgplot.

**Reply by walkloud** [reply]
Thanks for the info, I'll see what my colleague decides.

**Comment by teuben** [comment]
I don't understand the licensing issues, apart from the code not having been touched by the author in 20 or so years, the community has provided several patches to enable 64bit and other software rot, and all we get back in caltech licensing issues?   How come I can freely use it on debian and redhat flavors? (although the two have annoying differences in their enabled drivers)
If it's truly a licensing issue, Caltech should be embarrassed by holding such legacy code in an impossible ransom situation. Surely there must be a misunderstanding. Has somebody really spoken to a librarian or lawyer at caltech? 

I've tried both giza and plplot,and they won't work without major investments. I use it in legacy code, and I'd like to keep it that way.

**Reply by carlocab** [reply]
pgplot is in the non-free archive at Debian. See https://packages.debian.org/sid/source/pgplot5. So Debian certainly thinks the license is non-free, and this is a [requirement](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae) for distribution through Homebrew-owned repositories.

I kept an old version of the formula in [my tap](https://github.com/carlocab/homebrew-personal), which you can `brew install carlocab/personal/pgplot`. I don't test that it works, though, and there are no bottles (pre-built binaries) so it will build from source on installation. If that doesn't suit your needs you may wish to either host a version that does in your own tap, or otherwise build it from source yourself.

**Reply by teuben** [reply]
thanks for clarifying! I'll work on caltech  :-)   My first attempt on your tap didn't work on my linuxbrew, but that was just a test. On my catalina (in a QEMU virtual box) I have it mostly working, right now the pgxwin_server doesn't want to start, because I didn't put X on it yet. But the rest seems to work!!

**Reply by kazuakiyama** [reply]
Hi, I'm no longer an active user of PGPLOT, but I'm very slowly maintaining a local repository for a brew formula for my folks in the EHT Collaboration using Caltech's DIFMAP software package relying on PGPLOT. It is based on the original brew formula but includes some updates for macOS after Monterey. 
https://github.com/kazuakiyama/homebrew-pgplot
